### PR TITLE
refactor workflows

### PR DESCRIPTION
**Pull request by lexicoder**

### CODE REVIEW

File: .github/workflows/build_container.yml [hunk 1]
```diff
@@ -121,3 +119,15 @@ jobs:
...
+    runs-on: ubuntu-latest
+    needs: build_container
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: Run macOS Universal pkg Release workfow
+        env:
+          GH_TOKEN: ${{ secrets.REPO_API_TOKEN }}
+        run: gh workflow run pkg_macos.yaml
```
Comment (benr) [inline .github/workflows/build_container.yml:133]: We should call flows using peter-evans/repository-dispatch@v3 to be consistent with how we do it elsewhere.

**Comment by benr** [comment]
We need to create a 'release' for each release; the triggers shouldn't change for most workflows.

**Comment by lexicoder** [comment]
> We need to create a 'release' for each release; the triggers shouldn't change for most workflows.

I have refactored `release.yml` to create a release before calling the other workflows.

Regarding triggers for the other workflow, these have to change since we want to have coordinated release process that results in triggering the integration tests after the installer workflows runs are successful. Leaving the `on: types: [released]` and `on: types: [published]` triggers as-is would require us to have way of knowing when these runs are complete and successful of which there no elegant solutions to this.

One potential solution might be the `on: workflow_run` but this triggers regardless of success or failure of the preceding workflow so I don't think it is a good option.

File: .github/workflows/build_container.yml (hunk 1 above)
Comment (benr) [reply .github/workflows/build_container.yml]: I accept this, it is much cleaner, just not consistent.  We should perhaps adopt this method going forward elsewhere.

**Review by benr — APPROVED** [review]

**Review by benr — APPROVED** [review]

**Pull request merged**